synapse/telepath.py

```diff
@@ -969,17 +971,22 @@ async def _teleLinkLoop(self):
 
     async def proxy(self, timeout=10):
         await self.waitready(timeout=timeout)
-        return self._t_proxy
+        ret = self._t_proxy
+        if ret is None or ret.isfini is True:
```

Comment:
I think it'd be most appropriate to raise IsFini() if self.isfini here ( which should be true if waitready is returning early )